### ISSUE TITLE
Videos in playlist getting duplicate

### ISSUE THREAD

**Issue by armin567ip** [iv-org/invidious#4080]
Hi. let's say THIS video is in my playlist. and if I accidently click on adding to that playlist again. it's gonna get there again. which is a bug and it should only be there once or like display an error message that video already in playlist

**Comment by unixfox** [comment]
To be fair, it's a really minor bug, just delete the duplicates yourself. It's probably because there is no check if a video already exists.

**Comment by armin567ip** [comment]
> To be fair, it's a really minor bug, just delete the duplicates yourself. It's probably because there is no check if a video already exists.

Well I have an idea for you in how to get it fixed in the code.
You have to assign an ID for every video/shorts that is added to a playlist. And then for every video it has to check
If (id exists) then: error message
Else:
Add it to playlist.
Something like this. You got the idea

**Comment by k0zyrev** [comment]
This can't even be considered a bug. It's a feature. Say, you're creating a music playlist and you want one particular song to play after every second unique song in the playlist. Or maybe you want it to repeat exactly fifteen times and then go on to other songs. With your proposition it would be impossible.

**Comment by SamantazFox** [comment]
Yeah, I see no point in adding such a restriction. It complicates the code, and add no real benefits. I'm not against adding some logic to highlight duplicates, but that'll likely vome at the cost of making large amounts of DB queries (either one large `SELECT` or multiple `SELECT count(id)`).

**Comment by ZIOOIZ8** [comment]
First of all, many thanks to all developers for these great possibilities that Invidious offers and that it exists at all. My suggestion on the subject would be to add an option in the settings where you can choose whether duplicates are allowed or not (I wouldn't link this to IDs, but simply to the name of the videos). For example, I like to create playlists so that I can organize videos by topic and refer back to them if I'm confronted with the same or similar topics again months later. It makes it a lot easier to find certain videos in your own playlists than if you have to search the whole of YT for them. However, over time such lists fill up and unfortunately there is no way to sort them, this function alone would help a lot to find duplicates and delete them yourself.
